modules/ROOT/pages/dataweave-cookbook-merge-multiple-payloads.adoc

```diff
@@ -1,78 +1,23 @@
-= Merge Multiple Payloads
+= Merge Fields from Separate Objects
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 :keywords: studio, anypoint, transform, transformer, format, aggregate, filter, json, metadata, dataweave, data weave, datamapper, dwl, dfl, dw, output structure, input structure, map, mapping, contains, as
 
-This DataWeave example deals with an input that contains a collection of collections. The first payload index `payload[0]` lists price by book, and the second `payload[1]` lists authors by book, where each book can be identified by a unique `bookId` key. This transformation merges both payload array elements into one, matching their keys, where one book may have several authors.
+The DataWeave examples merge fields from separate input arrays. The first   (`firstInput`) is a DataWeave variable that lists price by book ID, and the second (`secondInput`) lists authors by book ID. Each book has a unique `bookId` key.
 
+The DataWeave scripts produce the same output. Both scripts use two `map` functions with a filter, and one of them also creates an alias for the `bookId`:
+* The first `map` function iterates over the elements of the `firstInput`
+array. After the function evaluates a given `price` field in `firstInput`, a second `map` function uses a filter to identify elements in the second payload (`secondInput`) that match the `filter` criteria (`secondInput filter ($.*bookId contains id)`). The filter returns an element in `secondInput` that contains a `bookId` value that matches the `bookId` value in the `firstInput` element.
```

Comment:
Why mention `price` if `map` is unrelated? It just happens to be after it.

Reply:
Not sure if people understand that the evaluation takes place during each iteration of `map` over elements in `firstInputValue`. I'll try to revise it a bit.

Reply:
revised